sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsTestBase.java

```diff
@@ -168,32 +175,45 @@ protected void createUsers(HttpClient httpClient) {
         secondParticipantId = communicationClient.createUser();
         thirdParticipantId = communicationClient.createUser();
 
-        firstParticipant = new RoomParticipant().setCommunicationIdentifier(firstParticipantId).setRole(RoleType.ATTENDEE);
-        secondParticipant = new RoomParticipant().setCommunicationIdentifier(secondParticipantId).setRole(RoleType.ATTENDEE);
-        thirdParticipant = new RoomParticipant().setCommunicationIdentifier(thirdParticipantId).setRole(RoleType.CONSUMER);
+        firstParticipant = new RoomParticipant(firstParticipantId);
+        secondParticipant = new RoomParticipant(secondParticipantId);
+        thirdParticipant = new RoomParticipant(thirdParticipantId, ParticipantRole.CONSUMER);
+
+
+        firstParticipant2 = new RoomParticipant(firstParticipantId, null);
 
-        firstChangeParticipant = new RoomParticipant().setCommunicationIdentifier(firstParticipantId).setRole(RoleType.CONSUMER);
-        secondChangeParticipant = new RoomParticipant().setCommunicationIdentifier(firstParticipantId).setRole(RoleType.CONSUMER);
+        secondParticipant2 = new RoomParticipant(secondParticipantId, null);
 
-        validateParticipant1 = new RoomParticipant().setCommunicationIdentifier(firstParticipantId);
-        validateParticipant2 = new RoomParticipant().setCommunicationIdentifier(secondParticipantId);
-        validateParticipant3 = new RoomParticipant().setCommunicationIdentifier(thirdParticipantId);
+        thirdParticipantWithDefaultRole = new RoomParticipant(thirdParticipantId, ParticipantRole.ATTENDEE);
+
+        firstChangeParticipant = new RoomParticipant(firstParticipantId, ParticipantRole.CONSUMER);
+        secondChangeParticipant = new RoomParticipant(secondParticipantId, ParticipantRole.CONSUMER);
+
+        validateParticipant1 = new RoomParticipant(firstParticipantId, ParticipantRole.ATTENDEE);
+        validateParticipant2 = new RoomParticipant(secondParticipantId, ParticipantRole.ATTENDEE);
+        validateParticipant3 = new RoomParticipant(thirdParticipantId, ParticipantRole.ATTENDEE);
 
         participants1 = Arrays.asList(firstParticipant, secondParticipant, thirdParticipant);
         participants2 = Arrays.asList(firstParticipant, secondParticipant);
         participants3 = Arrays.asList(secondParticipant);
         participants4 = Arrays.asList(firstParticipant, secondParticipant);
         participants5 = Arrays.asList(firstParticipant, secondParticipant, thirdParticipant);
+        participants6 = Arrays.asList(firstParticipant2, secondParticipant2);
```

Comment:
participants6 is overwritten on line 205? Again, let's just refactor this code to avoid the confusion

Reply:
Was added as a test line, removed now.